### PR TITLE
add a test for lib/wsapi/account_cancel

### PR DESCRIPTION
**Pull request by jrgm**
Obviously repeating a lot of setup vows that would be nice to have as something like `utils.addBatchAddressInfo(email, expect ...)`. 

Anyways, this wsapi lacked a backend test so here's mine.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
After rebasing off of dev, this test fails using both the JSON and mysql adapters.

```
➜  browserid-remotes git:(wsapi-account-cancel-test) NODE_ENV=test_mysql vows tests/account-cancel-test.js
···✗··············✗···✗··  


    the test user 
      ✗ is not a known user 
        » expected false, 
    got  undefined (===) // account-cancel-test.js:55 

    GET list_emails 
      ✗ returns an object with proper email 
        » expected 'someuser@somedomain.com', 
    got  'success' (===) // account-cancel-test.js:159 

    the test user 
      ✗ is not a known user after account was cancelled 
        » expected false, 
    got  undefined (===) // account-cancel-test.js:191 
  ✗ Broken » 22 honored ∙ 3 broken (4.972s) 
  %                                                                                 
➜  browserid-remotes git:(wsapi-account-cancel-test) NODE_ENV=test_json vows tests/account-cancel-test.js
···✗··············✗···✗··  


    the test user 
      ✗ is not a known user 
        » expected false, 
    got  undefined (===) // account-cancel-test.js:55 

    GET list_emails 
      ✗ returns an object with proper email 
        » expected 'someuser@somedomain.com', 
    got  'success' (===) // account-cancel-test.js:159 

    the test user 
      ✗ is not a known user after account was cancelled 
        » expected false, 
    got  undefined (===) // account-cancel-test.js:191 
  ✗ Broken » 22 honored ∙ 3 broken (4.801s) 
  %                                                                                 
➜  browserid-remotes git:(wsapi-account-cancel-test) 
```

**Comment by jrgm** [comment]
I updated the test and it passes locally for me. Travis completed the front and back_mysql but the vm seems to have just stalled in the middle of the back (json) test and timed out.

**Comment by jrgm** [comment]
Re-triggered the build with the next travis-ci system. All green now.

File: tests/account-cancel-test.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+var suite = vows.describe('account-cancel');
+
+// disable vows (often flakey?) async error behavior
+suite.options.error = false;
+
+start_stop.addStartupBatches(suite);
+
+const
+TEST_EMAIL = secrets.weakGenerate(12) + '@somedomain.com',
```
Comment (lloyd) [inline tests/account-cancel-test.js:40]: Nice way to generate ephemeral emails.

File: tests/account-cancel-test.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    "Content-type is 'application/json'": function(err, r) {
+      assert.strictEqual(r.headers['content-type'].indexOf('application/json'), 0);
+    },
+    "returns json string with { status: 'complete', userid: \\d+ }": function (err, r) {
+      assert.strictEqual(JSON.parse(r.body).status, "complete");
+      assert(/\d+/.test(JSON.parse(r.body).userid));
+    },
+  }
+});
+
```
Comment (lloyd) [inline tests/account-cancel-test.js:116]: All this setup stuff is orthogonal to what you are testing, and can use lib/secondary.js - not a merge blocker

**Comment by lloyd** [comment]
R+

File: tests/account-cancel-test.js (hunk 2 above)
Comment (jrgm) [inline tests/account-cancel-test.js:116]: Yeah, duh. Earlier this morning I was intending to do that, and then completely blanked on doing it. I will change this.

**Comment by lloyd** [comment]
R+ love it.  BTW, working on much better error handling in secondary.js in primary-transition-tests

**Comment by jrgm** [comment]
Yeah, I was looking at that and noticed it wasn't checking the response code (or content-type). /me tosses his branch ;-)

File: tests/account-cancel-test.js (hunk 2 above)
Comment (jrgm) [inline tests/account-cancel-test.js:116]: Did that. https://github.com/mozilla/browserid/pull/2830
